ESSArch_Core/maintenance/views.py

```diff
@@ -133,16 +123,22 @@ def preview(self, request, pk=None):
         for ip in ips:
             datadir = os.path.join(ip.policy.cache_storage.value, ip.object_identifier_value)
             for pattern, spec in job.rule.specification.items():
-                for path in iglob(datadir + '/' + pattern):
-                    if os.path.isdir(path):
-                        for root, dirs, files in walk(path):
-                            rel = os.path.relpath(root, datadir)
+                files = find_all_files(datadir, ip, pattern)
```

Comment:
The list must be updated, not replaced, on each iteration of the rule specification

```suggestion
                files.extend(find_all_files(datadir, ip, pattern))
```